nssp_demo/prep_data.py

```diff
@@ -8,43 +8,126 @@
+) -> pl.DataFrame:
+    if state_level_nssp_data is None:
+        return pl.DataFrame(
+            schema={
+                "date": pl.Date,
+                "geo_value": pl.Utf8,
+                "disease": pl.Utf8,
+                "ed_visits": pl.Float64,
+            }
+        )
+
     disease_map = {
         "COVID-19": "COVID-19/Omicron",
-        "Influenza": "Influenza",
-        "RSV": "RSV",
-        "Total": "Total",
     }
+    disease_key = disease_map.get(disease, disease)
+
+    return (
+        state_level_nssp_data.filter(
+            pl.col("disease").is_in([disease_key, "Total"]),
+            pl.col("metric") == "count_ed_visits",
+            pl.col("geo_value") == state_abb,
+            pl.col("geo_type") == "state",
+            pl.col("reference_date") >= first_training_date,
+            pl.col("report_date") == state_level_report_date,
+        )
+        .select(
+            [
+                pl.col("reference_date").alias("date"),
+                pl.col("geo_value").cast(pl.Utf8),
+                pl.col("disease").cast(pl.Utf8),
+                pl.col("value").alias("ed_visits"),
+            ]
+        )
+        .with_columns(
+            disease=pl.col("disease")
+            .cast(pl.Utf8)
+            .replace({v: k for k, v in disease_map.items()}),
+        )
+        .sort(["date", "disease"])
+        .collect()
+    )
 
-    data_to_save = (
-        nssp_data.filter(
-            (pl.col("disease").is_in([disease_map[disease], "Total"]))
-            & (pl.col("metric") == "count_ed_visits")
-            & (pl.col("geo_value") == state_abb)
-            & (pl.col("reference_date") >= first_training_date)
+
+def aggregate_facility_level_nssp_to_state(
+    facility_level_nssp_data: pl.LazyFrame,
+    state_abb: str,
+    disease: str,
+    first_training_date: str,
+) -> pl.DataFrame:
+    if facility_level_nssp_data is None:
+        return pl.DataFrame(
+            schema={
+                "date": pl.Date,
+                "geo_value": pl.Utf8,
+                "disease": pl.Utf8,
+                "ed_visits": pl.Float64,
+            }
+        )
+
+    disease_map = {
+        "COVID-19": "COVID-19/Omicron",
+    }
+    disease_key = disease_map.get(disease, disease)
+
+    return (
+        facility_level_nssp_data.filter(
+            pl.col("disease").is_in([disease_key, "Total"]),
+            pl.col("metric") == "count_ed_visits",
+            pl.col("geo_value") == state_abb,
+            pl.col("reference_date") >= first_training_date,
         )
         .group_by(["reference_date", "disease"])
         .agg(pl.col("value").sum().alias("ed_visits"))
         .with_columns(
-            pl.when(pl.col("reference_date") <= last_training_date)
-            .then(pl.lit("train"))
-            .otherwise(pl.lit("test"))
-            .alias("data_type")
+            disease=pl.col("disease")
+            .cast(pl.Utf8)
+            .replace({v: k for k, v in disease_map.items()}),
```

Comment:
I think you can just use `.replace(disease_map)`

From [polars docs](https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.replace.html):

```python
mapping = {2: 100, 3: 200}
df.with_columns(replaced=pl.col("a").replace(mapping))
```

Reply:
I am inverting the `disease_map` here (note `{v: k for v, v}`). Perhaps for readability I should define the inverse right along with the mapn and then call this with `replace(inverse_disease_map)`

Reply:
Done in 7bb74d7